build/i2c/src/lib.rs

```diff
@@ -218,6 +221,22 @@ struct I2cSensors {
     names: Option<Vec<String>>,
 }
 
+impl I2cSensors {
+    /// Checks whether two sensor sets are compatible
+    ///
+    /// "Compatible" means that they have the same number of sensors in each
+    /// category, meaning they can be represented by the same `struct`.
+    fn is_compatible_with(&self, other: &Self) -> bool {
+        self.temperature == other.temperature
```

Comment:
This might be too nitpicky so feel free to ignore, but: I slightly worry about a future change adding a field to `I2cSensors` and forgetting to update this method. It'd be more verbose, but we could do something like

```rust
let &Self {
    temperature,
    power,
    // etc
} = self;
let &Self {
    temperature: other_temperature,
    power: other_power,
    // etc
} = other;

temperature == other_temperature && power == other_power // etc
```

to force the destructuring to turn into a compile error if fields are added. (It would also be a convenient place to put a `// names are intentionally ignored` kind of comment, since both would end up with a `names: _` entry.)

Reply:
Yup, although I only need to unpack one of the values (not both).  Done in 76338f818c88bf1c24ed42ae0c704faddc7c1bb4